src/utils/pickItem/pickContainer.ts

```diff
@@ -0,0 +1,94 @@
+import { RevisionItem } from "../../tree/revisionManagement/RevisionItem";
+import { localize } from "../localize";
+import { pickContainerApp } from "./pickContainerApp";
+import { type RevisionDraftPickItemOptions } from "./PickItemOptions";
+import { getPickRevisionDraftStep, getPickRevisionStep, getPickRevisionsStep } from "./pickRevision";
+
+function getPickContainerStep(): AzureWizardPromptStep<QuickPickWizardContext> {
+    return new ContextValueQuickPickStep(ext.rgApiV2.resources.azureResourceTreeDataProvider, {
+        contextValueFilter: { include: ContainerItem.contextValueRegExp },
+        skipIfOne: true,
+    }, {
+        placeHolder: localize('selectContainer', 'Select a container'),
+    });
+}
+
+function getPickContainersStep(): AzureWizardPromptStep<QuickPickWizardContext> {
+    return new ContextValueQuickPickStep(ext.rgApiV2.resources.azureResourceTreeDataProvider, {
+        contextValueFilter: { include: ContainersItem.contextValueRegExp },
+        skipIfOne: true,
+    });
+}
+
+export async function pickContainer(context: IActionContext, options?: RevisionDraftPickItemOptions): Promise<ContainerItem> {
+    const containerAppItem: ContainerAppItem = await pickContainerApp(context);
+    return await runQuickPickWizard(context, {
+        promptSteps: getPickContainerSteps(containerAppItem, { autoSelectDraft: options?.autoSelectDraft }),
+        title: options?.title,
+    }, containerAppItem);
+}
+
+/**
+ * Assumes starting from the ContainerAppItem
+ */
+export function getPickContainerSteps(containerAppItem: ContainerAppItem, options?: RevisionDraftPickItemOptions): AzureWizardPromptStep<QuickPickWizardContext>[] {
+    const promptSteps: AzureWizardPromptStep<QuickPickWizardContext>[] = [];
+    if (containerAppItem.containerApp.revisionsMode === KnownActiveRevisionsMode.Multiple) {
+        promptSteps.push(getPickRevisionsStep());
+
+        if (options?.autoSelectDraft && ext.revisionDraftFileSystem.doesContainerAppsItemHaveRevisionDraft(containerAppItem)) {
+            promptSteps.push(getPickRevisionDraftStep());
+        } else {
+            promptSteps.push(getPickRevisionStep());
+        }
+    }
+
+    promptSteps.push(new ContainerItemPickSteps());
+    return promptSteps;
+}
+
+export class ContainerItemPickSteps<T extends AzureResourceQuickPickWizardContext> extends AzureWizardPromptStep<T> {
+    public async prompt(): Promise<void> {
+        // Nothing to prompt, just need to use the subwizard
+    }
+
+    public shouldPrompt(): boolean {
+        return false;
+    }
+
+    public async getSubWizard(context: T): Promise<IWizardOptions<T> | undefined> {
+        const lastNode: unknown = context.pickedNodes.at(-1);
+
+        let containers: Container[] = [];
+        if (ContainerAppItem.isContainerAppItem(lastNode)) {
+            const node = lastNode as ContainerAppItem;
+            containers = node.containerApp.template?.containers ?? [];
```

Comment:
I think for all of these, you can just do `lastNode.containerApp.template?.containers` since your static methods are asserting all of the types.

![{E5402CBF-9169-4817-B443-1F05E31010EF}](https://github.com/user-attachments/assets/8cc74199-3094-42a2-9e42-0723847bd26d)

Reply:
Oh true, I don't know why I thought I needed to assert the types like that...